metricflow-semantics/metricflow_semantics/sql/sql_exprs.py

```diff
@@ -1719,3 +1793,158 @@ def lineage(self) -> SqlExpressionTreeLineage:  # noqa: D102
+    else_expr: Optional[SqlExpressionNode]
+
+    @staticmethod
+    def create(  # noqa: D102
+        when_to_then_exprs: Dict[SqlExpressionNode, SqlExpressionNode], else_expr: Optional[SqlExpressionNode] = None
+    ) -> SqlCaseExpression:
+        parent_nodes: Tuple[SqlExpressionNode, ...] = ()
+        for when, then in when_to_then_exprs.items():
+            parent_nodes += (when,)
+            parent_nodes += (then,)
+
+        if else_expr:
+            parent_nodes += (else_expr,)
+
+        return SqlCaseExpression(parent_nodes=parent_nodes, when_to_then_exprs=when_to_then_exprs, else_expr=else_expr)
+
+    @classmethod
+    def id_prefix(cls) -> IdPrefix:  # noqa: D102
+        return StaticIdPrefix.SQL_EXPR_CASE_PREFIX
+
+    def accept(self, visitor: SqlExpressionNodeVisitor[VisitorOutputT]) -> VisitorOutputT:  # noqa: D102
+        return visitor.visit_case_expr(self)
+
+    @property
+    def description(self) -> str:  # noqa: D102
+        return "Case expression"
+
+    @property
+    def displayed_properties(self) -> Sequence[DisplayedProperty]:  # noqa: D102
+        return super().displayed_properties
+
+    @property
+    def requires_parenthesis(self) -> bool:  # noqa: D102
+        return False
+
+    @property
+    def bind_parameter_set(self) -> SqlBindParameterSet:  # noqa: D102
+        return SqlBindParameterSet()
+
+    def __repr__(self) -> str:  # noqa: D105
+        return f"{self.__class__.__name__}(node_id={self.node_id})"
+
+    def rewrite(  # noqa: D102
+        self,
+        column_replacements: Optional[SqlColumnReplacements] = None,
+        should_render_table_alias: Optional[bool] = None,
+    ) -> SqlExpressionNode:
+        return SqlCaseExpression.create(
+            when_to_then_exprs={
+                when.rewrite(column_replacements, should_render_table_alias): then.rewrite(
+                    column_replacements, should_render_table_alias
+                )
+                for when, then in self.when_to_then_exprs.items()
+            },
+            else_expr=(
+                self.else_expr.rewrite(column_replacements, should_render_table_alias) if self.else_expr else None
+            ),
+        )
+
+    @property
+    def lineage(self) -> SqlExpressionTreeLineage:  # noqa: D102
+        return SqlExpressionTreeLineage.merge_iterable(
+            tuple(x.lineage for x in self.parent_nodes) + (SqlExpressionTreeLineage(other_exprs=(self,)),)
+        )
+
+    def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D102
+        if not isinstance(other, SqlCaseExpression):
+            return False
+        return self.when_to_then_exprs == other.when_to_then_exprs and self.else_expr == other.else_expr
+
+
+class SqlArithmeticOperator(Enum):
+    """Arithmetic operator used to do math in a SQL expression."""
+
+    ADD = "+"
+    SUBTRACT = "-"
+    MULTIPLY = "*"
+    DIVIDE = "/"
+
+
+@dataclass(frozen=True, eq=False)
+class SqlArithmeticExpression(SqlExpressionNode):
```

Comment:
For potentially cleanup, seems like this can replace `SqlRatioComputationExpression` unless we want to explicitly have a node for that. But if we wanted to do that, i don't see why we wouldn't want to do it for other operations.

Reply:
I considered that but decided not to because the ratio expression has a bunch of extra logic related to casting & `NULLIF`, and I wasn't sure we would always want that for this node!